### PR TITLE
Add YOURLS Support

### PR DESCRIPTION
**Pull request by auisgold**
## Description
Added YOURLS as a link shortening option, use yourls in config with Provider URL being your URL for YOURLS and Provide Key being YOURLS signature

## Motivation and Context
Adds another popular URL shortener option that was used in WDR. 

## How Has This Been Tested?
Hasn't been tested, but request code is exact same as WDR, so if you have YOURLS setup, should work no issues. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ X ] My code follows the code style of this project.
- [ X ] My change requires a change to the documentation.
- [ X ] I have updated the documentation accordingly.
Option for yourls has been added to comment in config

### CODE REVIEW

File: src/controllers/controller.js [hunk 1]
```diff
@@ -60,6 +60,9 @@ class Controller extends EventEmitter {
 			case 'shlink': {
 				return new ShlinkUriShortener(this.log, this.config.general.shortlinkProviderURL, this.config.general.shortlinkProviderKey, this.config.general.shortlinkProviderDomain)
 			}
+			case 'yourls': {
+				return new YourlsUriShortener(this.log, this.config.general.shortlinkProviderURL, this.config.general_shortLinkPreoviderKey)
```
Comment (jfberry) [inline src/controllers/controller.js:64]: `general.shortLinkPreoviderKey`

File: src/lib/yourlsUrlShortener.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        } catch (error) {
+            if (error.response) {
+                console.log(error.response.data);
+                if ('message' in error.response.data){
+                    if (error.response.data.message.includes('already exists')){
+                        return error.response.data.shorturl;
+                    }
+                }
+                else{
+                    console.log(`Shortener[yourls]: Failed to shorten. Got ${error.response.status}. Error: ${error.response.data ? error.response.data.reason : '?'}.`)
```
Comment (jfberry) [inline src/lib/yourlsUrlShortener.js:29]: Should be to log `this.log.warn` (and below)

**Comment by jfberry** [comment]
This looks ready to go - please DM me in discord to confirm status with regard to testing

**Comment by jfberry** [comment]
requires merge to develop or rebase

**Comment by auisgold** [comment]
Ready to go

**Pull request merged**